### PR TITLE
Fix container no ipv6 when run container on --network=host

### LINKED ISSUE #31473: Endless number of messages "Firewalld running: false " in docker log
I run the new Docker version 17.03.0-ce, build 60ccb22 in swarm mode on Ubuntu 14.04: 
1 manager + 3 additional nodes. Everything works fine so far, but i see endless messages in `/var/log/upstart/docker.log` on all nodes including manager:
`INFO[0000] Firewalld running: false`                     
`INFO[0000] Firewalld running: false` 
`INFO[0000] Firewalld running: false` 
`INFO[0000] Firewalld running: false` 
`INFO[0000] Firewalld running: false` 
`INFO[0000] Firewalld running: false`

### LINKED ISSUE #32433: NO more ipv6 on all interfaces on Fedora 25 package docker-engine-17.04.0.ce-1.fc25.x86_64
**Description**
NO IPv6 address is available since docker-engine-17.04.0.ce-1.fc25.x86_64, not even on loop.

**Steps to reproduce the issue:**
1. On Fedora 25 x86_64: Update docker-engine to docker-engine-17.04.0.ce-1.fc25.x86_64
2. Docker run some image
3. No IPv6 address is available regardless of the interface

**Describe the results you received:**
Example:
```shell
root@mycont1:/# ip a
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
13: eth0@if14: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default 
    link/ether 02:42:ac:11:00:02 brd ff:ff:ff:ff:ff:ff
    inet 172.17.0.2/16 scope global eth0
       valid_lft forever preferred_lft forever
root@mycont1:/#
```
In that example, I used ubuntu:14.04 official docker image.

**Describe the results you expected:**
To have inet6 entries when showing interfaces.  
Package downgrade gives my IPv6 addresses back: ``dnf downgrade docker-engine docker-engine-selinux``

Example:
```shell
docker run -t -i --rm --name "mycont1" -h "mycont1"  "ubuntu:14.04" /bin/bash
root@mycont1:/# ip a
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
15: eth0@if16: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default 
    link/ether 02:42:ac:11:00:02 brd ff:ff:ff:ff:ff:ff
    inet 172.17.0.2/16 scope global eth0
       valid_lft forever preferred_lft forever
    inet6 fe80::42:acff:fe11:2/64 scope link tentative 
       valid_lft forever preferred_lft forever
```

**Additional information you deem important (e.g. issue happens only occasionally):**
Downgrading package to **17.03.1.ce-1.fc25** fixes the issue

**Output of `docker version`:**

```
± docker version
Client:
 Version:      17.04.0-ce
 API version:  1.28
 Go version:   go1.7.5
 Git commit:   4845c56
 Built:        Mon Apr  3 18:06:13 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.04.0-ce
 API version:  1.28 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   4845c56
 Built:        Mon Apr  3 18:06:13 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

**Output of `docker info`:**

```
± docker info
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 2
Server Version: 17.04.0-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: 
containerd version: 422e31ce907fd9c3833a38d7b8fdd023e5a76e73
runc version: 9c2d8d184e5da67c95d601382adf14862e4f2228
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.10.8-200.fc25.x86_64
Operating System: Fedora 25 (Workstation Edition)
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 15.35GiB
Name: vincentg-lap2
ID: 5JJQ:CP3G:POLH:NLLI:WSSP:IENF:QAAA:Q3CJ:QDRZ:6TEJ:3H6O:TL6V
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
nothing that matters

### PR DESCRIPTION
**Pull request by aboch**
Fixes #32433
Fixes #31473

Fixes regression where running a container on network=host will disable ipv6 on the host.

### CODE REVIEW

File: integration-cli/docker_cli_daemon_test.go [hunk 1]
```diff
@@ -428,6 +428,22 @@ func (s *DockerDaemonSuite) TestDaemonIPv6FixedCIDRAndMac(c *check.C) {
 	c.Assert(strings.Trim(out, " \r\n'"), checker.Equals, "2001:db8:1::aabb:ccdd:eeff")
 }
 
+// TestDaemonIPv6HostMode checks that when the running a container with
+// network=host the host ipv6 addresses are not removed
+func (s *DockerDaemonSuite) TestDaemonIPv6HostMode(c *check.C) {
```
Comment (cpuguy83) [inline integration-cli/docker_cli_daemon_test.go]: This is a pretty specific test (and kind of malicious!).
I guess it's cool that it's there, but seems a bit excessive.

**Review by cpuguy83 — APPROVED** [review]
LGTM

Made a comment about the test. Not sure what others think.

**Comment by sanimej** [comment]
On the libnetwork side, networkNamespace.isDefault will be true for overlay network and `ingress_sbox` namespaces. But I don't think ipv6 being enabled is an issue for these namespaces. 

LGTM.

**Review by sanimej — APPROVED** [review]

**Comment by aboch** [comment]
@santhosh The `ingress-sbox` is not affected, because the `OptionUseDefaultSandbox()` is not passed during its creation. The host-mode overlay network netns is because of the poor `osl.NewSandbox()` interface, the overload of the `osCreate` field. But it has no effect because no interfaces are added to the overlay nw sandbox (via `AddInterface()`) and ipv6 is not supported anyway on overlay network.

Although it is not a priority, I agree we should extend the `osl.NewSandbox` API to not overload the `osCreate` flag.  Will push something after release. Thanks.

File: integration-cli/docker_cli_daemon_test.go (hunk 1 above)
Comment (mavenugo) [inline integration-cli/docker_cli_daemon_test.go]: @cpuguy83 Idk how this is malicious. The test lgtm.

**Review by mavenugo — APPROVED** [review]
LGTM

**Pull request merged**